api/src/main/java/org/ehcache/spi/LifeCycledAdapter.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.spi;
+
+/**
+ * @author Mathieu Carbou
+ */
+public abstract class LifeCycledAdapter implements LifeCycled {
```

Comment:
Missed it was in the API module.
I believe we do not want that on the basis of not having concrete classes in the API.
core would be a better choice.

@chrisdennis @alexsnaps @lorban @AbfrmBlr @albinsuresh @rishabhmonga opinions?